packages/tools/pluggable-widgets-tools/configs/rollup-plugin-widget-typing.js

```diff
@@ -0,0 +1,33 @@
+import { promises as fs } from "fs";
+import { extname, join } from "path";
+import { listDir } from "./shared";
+
+const { transformPackage } = require("../dist/typings-generator");
+
+export function widgetTyping({ sourceDir }) {
+    let firstRun = true;
+
+    return {
+        name: "widget-typing",
+        async options(options) {
+            // We have to run transformation before typescript starts its resolution cache =>
+            // before the first buildStart starts, because buildStart is a "parallel" hook
+            await runTransformation(sourceDir);
+            return options;
+        },
+        async buildStart() {
+            (await listDir(sourceDir))
+                .filter(path => extname(path) === ".xml")
```

Comment:
Why do we need to get all files with xml extension then add them as watch ?

Reply:
To regenerate the typings if the xml is changed

Reply:
yes